src/python/ensembl/io/genomio/seq_region/prepare.py

```diff
@@ -14,439 +14,21 @@
-    if molecule_location is None:
-        molecule_location = MOLECULE_LOCATION
-    seq_region = {}
-    # Set accession as the sequence region name
-    src = "RefSeq" if is_refseq else "GenBank"
-    accession_id = data.get(f"{src}-Accn", "")
-    if accession_id and (accession_id != "na"):
-        seq_region["name"] = accession_id
-    else:
-        logging.warning(f'No {src} accession ID found for {data["Sequence-Name"]}')
-        return {}
-    # Add synonyms
-    synonyms = []
-    for field, source in synonym_map.items():
-        if (field in data) and (data[field].casefold() != "na"):
-            synonym = {"source": source, "name": data[field]}
-            synonyms.append(synonym)
-    if synonyms:
-        synonyms.sort(key=lambda x: x["source"])
-        seq_region["synonyms"] = synonyms
-    # Add sequence length
-    field = "Sequence-Length"
-    if (field in data) and (data[field].casefold() != "na"):
-        seq_region["length"] = int(data[field])
-    # Add coordinate system and location
-    seq_role = data["Sequence-Role"]
-    # Scaffold?
-    if seq_role in ("unplaced-scaffold", "unlocalized-scaffold"):
-        seq_region["coord_system_level"] = "scaffold"
-    # Chromosome? Check location
-    elif seq_role == "assembled-molecule":
-        seq_region["coord_system_level"] = "chromosome"
-        location = data["Assigned-Molecule-Location/Type"].lower()
-        # Get location metadata
-        try:
-            seq_region["location"] = molecule_location[location]
-        except KeyError as exc:
-            raise UnknownMetadata(f"Unrecognized sequence location: {location}") from exc
-    else:
-        raise UnknownMetadata(f"Unrecognized sequence role: {seq_role}")
-    return seq_region
-
-
-def report_to_csv(report_path: PathLike) -> Tuple[str, Dict]:
-    """Returns an assembly report as a CSV string.
-
-    Args:
-        report_path: path to a seq_region file from INSDC/RefSeq
-
-    Returns:
-        The data as a string in CSV format, and the head metadata as a dictionary.
-
-    """
-    with open_gz_file(report_path) as report:
-        data = ""
-        metadata = {}
-        last_head = ""
-        for line in report:
-            # Ignore header
-            if line.startswith("#"):
-                # Get metadata values if possible
-                match = re.search("# (.+?): (.+?)$", line)
-                if match:
-                    metadata[match.group(1)] = match.group(2)
-                last_head = line
-            else:
-                if last_head:
-                    data += last_head[2:].strip() + "\n"
-                    last_head = ""
-                data += line
-        return data, metadata
-
-
 def prepare_seq_region_metadata(
     genome_file: PathLike,
     report_file: PathLike,
     dst_file: PathLike,
-    gbff_file: Optional[PathLike] = None,
-    brc_mode: bool = False,
-    to_exclude: Optional[List[str]] = None,
+    gbff_file: PathLike | None = None,
```

Comment:
I would replace `PathLike` by `ensembl.utils.StrPath`, as it is more flexible and generic.